src/t8_cmesh.h

```diff
@@ -632,14 +659,33 @@ t8_cmesh_get_tree_vertices (t8_cmesh_t cmesh, t8_locidx_t ltreeid);
  * \param [in]     key          A key used to identify the attribute under all
  *                              attributes of this tree with the same \a package_id.
  * \param [in]     tree_id      The local number of the tree.
- * \param [out]    data_size    The size of the attribute in bytes.
  * \return         The attribute pointer of the tree \a ltree_id or NULL if the attribute is not found.
- * \a cmesh must be committed before calling this function.
+ * \note \a cmesh must be committed before calling this function.
  * \see t8_cmesh_set_attribute
  */
 void *
 t8_cmesh_get_attribute (t8_cmesh_t cmesh, int package_id, int key, t8_locidx_t ltree_id);
 
+/** Return the attribute pointer of a tree for a gloidx_t array.
+ * \param [in]     cmesh        The cmesh.
+ * \param [in]     package_id   The identifier of a valid software package. \see sc_package_register
+ * \param [in]     key          A key used to identify the attribute under all
+ *                              attributes of this tree with the same \a package_id.
+ * \param [in]     tree_id      The local number of the tree.
+ * \param [in]     data_count   The number of entries in the array that are requested. 
+ *                              This must be smaller or equal to the \a data_count parameter
+ *                              of the corresponding call to \ref t8_cmesh_set_attribute_gloidx_array
+ * \return         The attribute pointer of the tree \a ltree_id or NULL if the attribute is not found.
+ * \note \a cmesh must be committed before calling this function.
+ * \note No check is performed whether the attribute actually stored \a data_count many entries since
+ *       we do not store the number of data entries of the attribute array.
+ *       You can keep track of the data count yourself by using another attribute.
+ * \see t8_cmesh_set_attribute_gloidx_array
+ */
+t8_gloidx_t *
+t8_cmesh_get_attribute_gloidx_array (t8_cmesh_t cmesh, int package_id, int key, t8_locidx_t ltree_id,
+                                     size_t data_count);
```

Comment:
```suggestion
                                     const size_t data_count);
```
It is possible to make ltree_id constant, too, but I propose to do this in another PR, since it requires changes outside of the topic of this PR